src/core/execute.c

```diff
@@ -2958,6 +2958,23 @@ static int exec_parameters_get_cgroup_path(const ExecParameters *params, char **
         return using_subcgroup;
 }
 
+static int exec_context_cpu_affinity_from_numa(const ExecContext *c, CPUSet *ret) {
+        assert(c);
+        assert(ret);
+
+        if (!c->derive_cpu_affinity_from_numa_mask)
+                return 0;
+
+        if (!c->numa_policy.nodes.set) {
+                log_warning("Can't derive CPU affinity mask from NUMA mask because NUMA mask is not set, ignoring");
+                return 0;
+        }
+
+        cpu_set_reset(ret);
+
+        return numa_to_cpu_set(&c->numa_policy, ret);
+}
```

Comment:
as a matter of coding style, we generally want functions that initialize all return parameters on success and leave them untouched on failure. This function leaves them uninitialized in two success cases, which we really should avoid... And it it might change it even on failure.